alphabet-soup/src/test/scala/io/typechecked/alphabetsoup/AtomMacroSpec.scala

```diff
@@ -152,6 +152,35 @@ class AtomMacroSpec extends FlatSpec with Matchers {
    illTyped("@Atomic object Foo",".*Invalid: Can not annotate structure with @Atomic.*")
 
   }
+  it should "create an implicit Atom for a class with a type parameter" in {
+    @Atomic class Foo[A]
+    implicitly[Atom[Foo[String]]]
+  }
+  it should "create an implicit Atom for a class with a higher-kinded type parameter" in {
+    @Atomic class Foo[A[_]]
+    implicitly[Atom[Foo[Option]]]
+  }
+  it should "create an implicit Atom for a class with many type parameters" in {
+    @Atomic class Foo[A[_], B, C]
+    implicitly[Atom[Foo[Option, String, (Boolean, List[String])]]]
+  }
+  it should "create an implicit Atom for a class with a type parameter with a type bound" in {
+    trait S
+    class T extends S
+    class U extends T
+    @Atomic class Foo[A <: T]
+    implicitly[Atom[Foo[U]]]
+    assertDoesNotCompile("implicitly[Atom[Foo[S]]]")
```

Comment:
Could you match on the compilation error message here? You might after all have just mis-matched the brackets, causing a compilation failure.

If `assertDoesNotCompile` doesn't let you do that then I know shapeless' `illTyped` does, and there should be example floating around for you to copy